types/database/database.d.ts

```diff
@@ -91,11 +91,12 @@ export type HandlePostsArgs = {
 export type HandleThreadsArgs = {
   threads: Thread[];
   prepareRecordsOnly?: boolean;
-  teamId: string;
+  teamId?: string;
```

Comment:
why is this optional? I thought every thread belongs to a team even for DM/GM's

Reply:
Yes, it does but we also save "not following" threads in the threads table and there is no need for them to exist in the team. They are only for the in-channel experience (replyCount and Participants list)

Reply:
this thing is driving me crazy, there seems to be a lot of definitions apply for the same thing but then they are different but can apply for multiple things.

"not following"  for me is just another field called `follow` that has nothing to do with the teamId, the schema does not define `team_id` as optional thus I expect to have it at all times. This will in turn avoid having queries that return unexpected data and so on.

Reply:
`Threads` table contains all the root posts "following" or "not following"
`ThreadsInTeam` contains all the "following" threads in the team

We can definitely save "not following" threads in `ThreadsInTeam` table but as we have no use for it in terms of the query, we thought of making it optional.

Reply:
alright.. I guess this thing has a lot of moving parts